apps/extension/src/background/keyring/keyring.ts

```diff
@@ -373,13 +379,9 @@ export class KeyRing {
       // Cloning the seed, otherwise it gets zeroized in deriveTransparentAccount
       const seedClone = seed.clone();
       const path = { account: 0, change: 0, index };
-      const accountInfo = this.deriveTransparentAccount(
-        seedClone,
-        path,
-        parentId
-      );
+      const accountInfo = deriveFn(seedClone, path, parentId);
       const balances: [string, string][] = await this.query.query_balance(
-        accountInfo.address
+        accountInfo.owner
```

Comment:
What does this change do?

Reply:
So this was the source of a bug that I mentioned. Basically for querying balances what we need to pass is an owner, which is either address for transparent accounts or extended viewing key for shielded.